test/puppetlabs/kitchensink/core_test.clj

```diff
@@ -204,6 +204,26 @@
       (testing "a weight is not numeric"
         (is (thrown? AssertionError (rand-weighted-selection :foo :bar)))))))
 
+(deftest rand-str-test
+  (testing "rand-str"
+    (testing "throws an IllegalArgumentException when given an unknown characters keyword"
+      (is (thrown? IllegalArgumentException (rand-str :CJK 42)))
+
+      (testing "and the exception's message mentions the given keyword"
+        (let [e (try (rand-str :CJK 42) (catch Exception e e))]
+          (is (re-find #":CJK" (.getMessage e))))))
```

Comment:
Could use `(is (thrown-with-msg? ...` to combine the assertions for the exception type and message together but separate seems fine as well.